src/instrumentsscene/view/instrumentsonscorelistmodel.cpp

```diff
@@ -321,16 +323,67 @@ const ScoreOrder& InstrumentsOnScoreListModel::currentScoreOrder() const
 
 void InstrumentsOnScoreListModel::onRowsMoved()
 {
-    const ScoreOrder& customOrder = notation::customOrder();
+    verifyScoreOrder();
+}
+
+void InstrumentsOnScoreListModel::onRowsRemoved()
+{
+    verifyScoreOrder();
+}
+
+bool InstrumentsOnScoreListModel::matchesScoreOrder() const
+{
+    QList<int> instrumentIndices;
+    for (const Item* item : items()) {
+        auto instrument = dynamic_cast<const InstrumentItem*>(item);
+        if (!instrument) {
+            continue;
+        }
+        instrumentIndices << currentScoreOrder().instrumentSortingIndex(instrument->id, instrument->isSoloist);
```

Comment:
Better to get this value outside of the loop and use it here (and in the return statement)

Reply:
I always hoped the compiler would take care of this :smile: